horace_core/algorithms/upgrade_legacy_alignment.m

```diff
@@ -0,0 +1,76 @@
+function out = upgrade_legacy_alignment(in_data,alatt0,angdeg0)
+% Change the crystal lattice and orientation of an sqw object stored in a file
+% or celarray of files
+%
+%   >> upgrade_legacy_alignment(filenames, alatt0,angdeg0)
+% modify legacy alignment previously
+%                                                 % of the crystal according to the
+%                                                 % crystal alignment information provided
+%
+%
+% Input:
+% -----
+%   w          cellarray of Input sqw objects or file names for sqw files,
+%              containing sqw objects, aligned using legacy alignment
+%
+%  alatt0   -- vector of lattice parameters for initial not aligned file
+%  angdeg0  -- vector of lattice angles for initial not aligned file.
+%
+
+% Output:
+% -------
+%   wout      cellarray of iutput sqw objects and filename containing sqw
+%             objects modified
+%
+
+% This routine used to change the crystal in sqw files, when it overwrites the input file.
+
+% Parse input
+% -----------
+if ischar(in_data) % process single file
+    in_data = {in_data};
+end
+if ~isnumeric(alatt0) || ~isnumeric(angdeg0) || numel(alatt0) ~= 3 || numel(angdeg0) ~= 3 || nargin < 3
+    error('HORACE:upgrade_legacy_alignment:invalid_argument',...
+        'To upgrade legacy alignment to modern format, one needs to provide lattice parameters for non-aligned lattice');
+end
+hc = hor_config;
+ll = hc.log_level;
+
+% Perform operations
+out = cell(1,numel(in_data));
+for i=1:numel(in_data)
+    if isa(in_data{i},'SQWDnDBase')
+        out{i} = in_data{i}.upgrade_legacy_alignment(alatt0,angdeg0);
+    else
```

Comment:
should test this is a filename

Reply:
faccessor is extensively testing its input and will throw if found something inappropriate.